components/tidb_query/src/rpn_expr/impl_time.rs

```diff
@@ -70,6 +70,22 @@ pub fn day_of_year(ctx: &mut EvalContext, t: &Option<DateTime>) -> Result<Option
     Ok(Some(Int::from(day)))
 }
 
+#[rpn_fn(capture = [ctx])]
+#[inline]
+pub fn to_days(ctx: &mut EvalContext, t: &Option<DateTime>) -> Result<Option<Int>> {
+    let t = match t {
+        Some(v) => v,
+        _ => return Ok(None),
+    };
+
+    if t.is_zero() {
```

Comment:
if `t.invalid_zero()`

Reply:
Hi, I use `t.invalid_zero()` in the first commit, but @breeswish said "The original implementation is if t.is_zero()", then I changeg it back. So maybe `t.is_zero()` is good enough?

Reply:
We can keep the original implementation for now and make a bulk change later. Or you can change both the original implementation and the vectorized implementation to a more correct one. Both are fine to me.

Reply:
Hi, I change both vectorized and un-vectorized implementation. 
And I find the definition of `invalid_zero` function is `self.month() == 0 || self.day() == 0`, so that is why the `invalid_zero` is a more correct one?